### PR TITLE
[sonic-cfggen]:move get_machine_info function to lib

### PR DESCRIPTION
**Pull request by sihuihan88**
Signed-off-by: Sihui Han <sihan@microsoft.com>

### CODE REVIEW

**Comment by jleveque** [comment]
Machine info is not minigraph-related, so it feels a bit out of place in minigraph.py.

Is there a need to call this function that can't be accomplished by calling sonic-cfggen?

**Comment by sihuihan88** [comment]
Machine info is actually used to parse the minigraph. The reason we move it here is that teamshow and maybe other tools in future will need to use this function to parse the minigraph. It's better to reference a python library rather than calling a script (sonic-cfggen) directly. That's the reason why we decide to move it here after discussion.

**Comment by qiluo-msft** [comment]
1. Machine info is *not* used to parse the minigraph.
2. We can keep them in sonic-cfggen, and we can import it as a library technically.

---
In reply to: [292682895](https://github.com/Azure/sonic-buildimage/pull/489#issuecomment-292682895) [](ancestors = 292682895)

**Comment by lguohan** [comment]
agree that get_platform_info can be a library, but it is different from minigraph, so can we have a separate library file for it? maybe sonic_platform.py?

**Comment by sihuihan88** [comment]
I think we need to use the machine info during parsing minigraph. Platform info determines the directory of port_config.ini which will be used to find the port_alias_map during parsing the minigraph. Without correct platform info, parsing minigraph will fail. Also as the API of parse_xml shows, it takes platform as input.

It also makes sense that we separate it from minigraph. I will create a new library file called sonic_machine.py will provides all the machine related info in the same folder.

**Comment by stcheng** [comment]
What about the file port_config.ini? Currently the parsing logic is inside the minigraph.py file although it is also not necessary for parsing the xml file. Do we want to treat the port_config.ini file similar to machine.conf file? Do we want to keep the port_config.ini parsing logics still inside the minigraph.py file or into a separate file or into the sonic_platform.py library?

**Comment by lguohan** [comment]
parse port_config.ini can also be moved to sonic_platform.py

**Comment by sihuihan88** [comment]
parse_port_config relies updates port_alias_map which is a global variable in minigraph.py during its process. Moving it out of minigraph.py might require major code change. The current motivation is to unblock teamshow. I will move the get_platform_info and get_machine_info to sonic_platform.py first. We can create an issue and refactor the parse_port_config part later.

**Comment by stcheng** [comment]
agree to refactor port_config.ini later.

**Comment by sihuihan88** [comment]
Updated as comments.

**Comment by taoyl-ms** [comment]
I think it makes sense to move port_config.ini reading logic out from minigraph.py and remove that global variable as well. I'm fine either doing it in this PR or I could do it later.

File: src/sonic-config-engine/setup.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def get_test_suite():
       author='Taoyu Li',
       author_email='taoyl@microsoft.com',
       url='https://github.com/Azure/sonic-buildimage',
-      py_modules=['minigraph', 'openconfig_acl'],
+      py_modules=['minigraph', 'openconfig_acl', 'sonic_platform'],
```
Comment (qiluo-msft) [inline src/sonic-config-engine/setup.py]: >'minigraph', 'openconfig_acl', 'sonic_platform' [](start = 18, length = 47)

How about just one module? @taoyl-ms  any comment?

Comment (taoyl-ms) [inline src/sonic-config-engine/setup.py]: I'm for splitting into multiple modules based on functionality. Actually I've always been feeling not comfortable to have port config logic in minigraph but have not had chance to refactor it.

**Comment by taoyl-ms** [comment]
In reply to @qiluo:

1. Machine info *IS* needed before parse the minigraph. Dependency is minigraph->port_config->machine_info
2. I think that as soon as a module is shared by more than one downstream, it should be taken out of script and encapsulated as a module. I don't think it is a right thing to reference functions in another *script*.

---
In reply to: [292685706](https://github.com/Azure/sonic-buildimage/pull/489#issuecomment-292685706) [](ancestors = 292685706,292682895)

**Comment by sihuihan88** [comment]
retest please

**Comment by stcheng** [comment]
@sihuihan88 could you add new issues mentioned in this pull request?

**Comment by sihuihan88** [comment]
@stcheng Just created.

**Pull request merged**